python/packages/azurefunctions/agent_framework_azurefunctions/_durable_agent_state.py

```diff
@@ -0,0 +1,1238 @@
+                contents.append(content_dict)  # type: ignore
+
+        return cls(
+            role=data.get("role", ""),
+            contents=contents,
+            author_name=data.get("authorName"),
+            created_at=created_at,
+            extensionData=data.get("extensionData"),
+        )
+
+    @property
+    def text(self) -> str:
+        """Extract text from the contents list."""
+        text_parts = []
+        for content in self.contents:
+            if isinstance(content, DurableAgentStateTextContent):
+                text_parts.append(content.text or "")
+        return "".join(text_parts)
+
+    @staticmethod
+    def from_chat_message(content: Any) -> DurableAgentStateMessage:
+        """Converts a ChatMessage or RunRequest from the agent framework to a DurableAgentStateMessage.
+
+        Args:
+            content: ChatMessage or RunRequest object with role, message/contents, and metadata
+
+        Returns:
+            DurableAgentStateMessage with converted content items and metadata
+        """
+        # Convert to a list of DurableAgentStateContent objects
+        contents_list: list[DurableAgentStateContent] = []
+
+        if hasattr(content, "message") and isinstance(content.message, str):
+            # RunRequest with 'message' attribute
+            contents_list = [DurableAgentStateTextContent(text=content.message)]
+        elif hasattr(content, "contents") and content.contents:
+            # ChatMessage with 'contents' attribute - convert each content object
+            for c in content.contents:
+                converted = DurableAgentStateContent.from_ai_content(c)
+                contents_list.append(converted)
+
+        # Convert role enum to string if needed
+        role_value = content.role.value if hasattr(content.role, "value") else str(content.role)
+
+        return DurableAgentStateMessage(
+            role=role_value,
+            contents=contents_list,
+            author_name=content.author_name if hasattr(content, "author_name") else None,
+            created_at=content.created_at if hasattr(content, "created_at") else None,
+            extensionData=content.extensionData if hasattr(content, "extensionData") else None,
+        )
+
+    def to_chat_message(self) -> Any:
+        """Converts this DurableAgentStateMessage back to an agent framework ChatMessage.
+
+        Returns:
+            ChatMessage object with role, contents, and metadata converted back to agent framework types
+        """
+        from collections.abc import MutableMapping, Sequence
+        from typing import cast
+
+        from agent_framework import ChatMessage
+
+        # Convert DurableAgentStateContent objects back to agent_framework content objects
+        ai_contents = [c.to_ai_content() for c in self.contents]
+
+        # Build kwargs for ChatMessage
+        kwargs: dict[str, Any] = {
+            "role": self.role,
+            "contents": cast(Sequence[Any], ai_contents),
```

Comment:
NIT: Unnecessary cast

```suggestion
            "contents": ai_contents,
```